lib/TPP/TestMatchers.cpp

```diff
@@ -159,6 +159,28 @@ void testTppIdentity(FunctionOpInterface funcOp) {
   });
 }
 
+void testTppRank(FunctionOpInterface funcOp) {
+  // clang-format off
+  auto matcher0 =
+    StructuredOpMatcher::make<linalg::GenericOp>()
+      .input(MatchAll(), HasRank({HasRank::SCALAR, 1}));
+  auto matcher1 =
+    StructuredOpMatcher::make<linalg::GenericOp>()
+      .input(MatchAll(), HasRank({2}));
```

Comment:
Why is `matcher1` actually matches `2` when it reports "match 1"?

Reply:
done.